src/server/lock_manager/deadlock.rs

```diff
@@ -924,6 +1316,66 @@ where
         }
     }
 
+    fn handle_replace_locks_by_keys_locally(&mut self, items: Vec<ReplaceLockByKeyItem>) {
+        let detect_table = &mut self.inner.borrow_mut().detect_table;
+        for item in items {
+            let deadlocks = detect_table.replace_lock_by_key(
+                item.key_hash,
+                item.key.clone(),
```

Comment:
It seems the `items` could be consumed in this function, perhaps we could save one `key.clone()`?

Reply:
To do this, we need to identify the last iteration of the following `for` loop, which makes the code ugly. So I didn't do this.
Actually, we may better consider to use something like `Arc<[u8]>` instead of `Vec<u8>` in the deadlock detector, as there is now too many cloning.